server/channels/app/scheduled_post_job.go

```diff
@@ -374,5 +375,44 @@ func (a *App) handleFailedScheduledPosts(rctx request.CTX, failedScheduledPosts
 			"scheduled_posts_failed",
 			map[string]any{"count": len(failedScheduledPosts)},
 		)
+		a.notifyUserAboutFailedScheduledMessages(rctx, failedScheduledPosts)
 	}
 }
+
+func (a *App) notifyUserAboutFailedScheduledMessages(rctx request.CTX, failedMessages []*model.ScheduledPost) {
+	a.Srv().Go(func() {
+		systemBot, err := a.GetSystemBot(rctx)
+		if err != nil {
+			rctx.Logger().Error("Failed to get the system bot", mlog.Err(err))
+			return
+		}
+
+		channel, err := a.GetOrCreateDirectChannel(rctx, failedMessages[0].UserId, systemBot.UserId)
+		if err != nil {
+			rctx.Logger().Error("Failed to get or create the DM", mlog.Err(err))
+			return
+		}
+
+		user, err := a.GetUser(failedMessages[0].UserId)
```

Comment:
We arrive here with the failed posts for just one user? Or we may receive the failed posts for several users?

Reply:
nice catch, for various users, will update and re-request review when ready! thanks @larkox 👍